doc/source/whatsnew/v1.1.0.rst

```diff
@@ -936,6 +936,7 @@ Reshaping
 - Bug in :meth:`DataFrame.replace` casts columns to ``object`` dtype if items in ``to_replace`` not in values (:issue:`32988`)
 - Ensure only named functions can be used in :func:`eval()` (:issue:`32460`)
 - Fixed bug in :func:`melt` where melting MultiIndex columns with ``col_level`` > 0 would raise a ``KeyError`` on ``id_vars`` (:issue:`34129`)
+- Fixed bug in :func:`merge` where an error was raised when performing an ``inner`` join with partial index and ``right_index`` when result was empty (:issue:`33814`)
```

Comment:
Small nits: don't need double back ticks around "inner", and technically the bug occurs when there's no overlap between the indexes rather than when the result DataFrame is empty (which could happen from having no columns for instance)

Reply:
Thank, did not know that with the back ticks. Changed the description

Reply:
I meant we actually don't need back ticks at all ("inner" is part of a phrase instead of being a keyword)

Reply:
Ah, I got it now I think